### PR TITLE
Added tests for UsersController

### PR DESCRIPTION
**Pull request by mrasync**
Hello Guys, 

I'm a new candidate in the recruitment process. I've created this PR to show my skills and improve this project :-)

In this PR I've added a couple tests for UsersController. The tests allow us to refactor (e.g replace promise calls with async/await syntax) without big risk of new errors.

Besides I've extracted some logic to helpers functions because controllers in my opinion should be as small as possible. Otherwise it will be hard to test in the future.

To run tests just type: `npm test`

### CODE REVIEW

File: server/controller/users.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
-import _ from "lodash";
 import Promise from "bluebird";
 
+import {collectValidUsers, collectUsersProperties} from "../helpers/users";
```
Comment (michaloo) [inline server/controller/users.js]: `npm run test:lint` would show some errors for these lines.

File: server/controller/users.js [hunk 2]
```diff
@@ -13,47 +17,13 @@ export default class UsersController {
    * @return {Promise}
    */
   sendUsersJob(req) {
-    const users = (req.payload.users || []).filter(u => !_.isEmpty(u.email));
-
+    const users = collectValidUsers(this._getUsers(req));
     if (users.length === 0) {
-      return req.hull.client.logger.log("skip psendUsersJob - empty users list");
+      return this._skip(req);
```
Comment (michaloo) [inline server/controller/users.js]: I like the idea of trimming down the action methods and extracting data related logic. But do you some other place where we can put these helper methods?

**Comment by michaloo** [comment]
Many thanks @mrasync for the PR.
I've marked it `changes requested` to start some discussion about placement for the helper methods.

File: server/controller/users.js (hunk 1 above)
Comment (mrasync) [inline server/controller/users.js]: U la la.. That's my mistake. I'll fix it.

File: server/controller/users.js (hunk 2 above)
Comment (mrasync) [inline server/controller/users.js]: Nope, I think we should wait for more logic to answer for this question :)

**Comment by sbellity** [comment]
@mrasync thanks for the PR !

**Comment by michaloo** [comment]
@mrasync - I would merge it right away, but before please take a look at our other ship here:
https://github.com/hull-ships/hull-mailchimp/tree/release/v0.1.0/server/jobs
As you can see we split controller classes intro separate functions - each in one file.

Do you see an opportunity to prepare to such switch with these helper methods?

File: server/controller/users.js (hunk 2 above)
Comment (michaloo) [inline server/controller/users.js]: What do you think about having it in such modules:
https://github.com/hull-ships/hull-hubspot/blob/develop/server/lib/hull-agent.js
https://github.com/hull-ships/hull-hubspot/blob/develop/server/lib/hubspot-agent.js

This services are meant as supersets of Hull and Hubspot APIs

**Comment by mrasync** [comment]
That sounds great! No problem.

But I'm not sure if I understand you well. Do you want to extract `sendUsersJob` & helpers to one module and `saveContactsJob` to other, right?

**Comment by michaloo** [comment]
Closing the PR since we are continuing the refactor on other branches.

**Pull request closed without merging**